tutorials/examples/import-demo-4.dart

```diff
@@ -0,0 +1,226 @@
+import "package:pointycastle/macs/hmac.dart";
+import "package:pointycastle/signers/rsa_signer.dart";
+import 'package:pointycastle/block/modes/cbc.dart';
+import 'package:pointycastle/paddings/pkcs7.dart';
+import 'package:pointycastle/random/fortuna_random.dart';
+*/
+import "package:pointycastle/asymmetric/api.dart";
+import "package:pointycastle/key_derivators/api.dart";
+import "package:pointycastle/key_generators/api.dart";
+
+void main() {
+  usingRegistry();
+  //explicit();
+}
+
+void usingRegistry() {
+  final sha256 = Digest("SHA-256");
+  final sha1 = Digest("SHA-1");
+  final md5 = Digest("MD5");
+
+  final _digest = sha256.process(_data);
+
+  final hmacSha256 = Mac("SHA-256/HMAC");
+  final hmacSha512 = Mac("SHA-512/HMAC");
+  final hmacMd5 = Mac("MD5/HMAC");
+
+  final _hmacValue = hmacSha256.process(_data);
+
+  final kd = KeyDerivator("SHA-256/HMAC/PBKDF2");
+
+  final _sGen = Random.secure();
+  final _seed =
+  Uint8List.fromList(List.generate(32, (n) => _sGen.nextInt(255)));
+  final secRnd = SecureRandom("Fortuna")..seed(KeyParameter(_seed));
+
+  // AES-CBC encryption
+
+  final _salt = secRnd.nextBytes(32);
+
+  final keyDerivator256 = KeyDerivator("SHA-256/HMAC/PBKDF2")
+    ..init(Pbkdf2Parameters(_salt, 10000, 256 ~/ 8));
+
+  final aes256key = keyDerivator256.process(_secret);
+
+  final _iv = secRnd.nextBytes(128 ~/ 8);
+  final aesCbc = BlockCipher("AES/CBC")
+    ..init(true, ParametersWithIV(KeyParameter(aes256key), _iv));
+
+  final _paddedData = Uint8List(
+      _data.length + (aesCbc.blockSize - (_data.length % aesCbc.blockSize)))
+    ..setAll(0, _data);
+  Padding("PKCS7").addPadding(_paddedData, _data.length);
+
+  final _ciphertext = aesCbc.process(_paddedData);
+
+  // RSA key generation and signing
+
+  final keyGen = KeyGenerator("RSA");
+  keyGen.init(ParametersWithRandom(
+      RSAKeyGeneratorParameters(BigInt.parse('65537'), 2048, 64), secRnd));
+  final _pair = keyGen.generateKeyPair();
+
+  final signer = Signer("SHA-256/RSA")
+    ..init(true, PrivateKeyParameter<RSAPrivateKey>(_pair.privateKey));
+
+  final _signature = signer.generateSignature(_data) as RSASignature;
+
+  final verifier = Signer("SHA-256/RSA")
+    ..init(false, PublicKeyParameter<RSAPublicKey>(_pair.publicKey));
+  final sigOk = verifier.verifySignature(_data, _signature);
+
+  print('''
+Data: "${utf8.decode(_data)}"
+
+SHA-256: ${bin2hex(_digest)}
+SHA-1:   ${bin2hex(sha1.process(_data))}
+MD5:     ${bin2hex(md5.process(_data), separator: ':')}
+
+HMAC-SHA256: ${bin2hex(_hmacValue)}
+HMAC-512:    ${bin2hex(hmacSha512.process(_data))}
+HMAC-MD5:    ${bin2hex(hmacMd5.process(_data))}
+
+AES-CBC ciphertext:
+${bin2hex(_ciphertext, wrap: 64)}
+
+Signature:
+${bin2hex(_signature.bytes, wrap: 64)}
+Verifies: $sigOk
+''');
+}
+
+//----------------------------------------------------------------
+
+final _data = utf8.encode('Hello world!');
+
+final _secret = utf8.encode('p@ssw0rd');
+
+/*
```

Comment:
Idem

Reply:
Same reason. Keeping the code so "diff" with the other import-demo-?.dart files is not cluttered, but if left in the file will not run/compile.

P.S. The variables starting with underscores are supposed to be the ones whose values aren't important to using Pointy Castle, and the ones without underscores are the ones that are important. For example, the `_secret` doesn't have to come from UTF decoding something.